### PR TITLE
Line ending fixes for json tests

### PR DESCRIPTION
**Pull request by krwq**
Fixes: https://github.com/dotnet/corefx/issues/38948

This makes it a bit less annoying to work with Json tests on Windows with line endings conversion enabled.

Currently getting around 35 failures after checking out

### CODE REVIEW

File: src/System.Text.Json/tests/JsonTestHelper.cs [hunk 1]
```diff
@@ -727,5 +727,12 @@ public static string GetCompactString(string jsonString)
 
         public static string StripWhitespace(this string value)
             => s_stripWhitespace.Replace(value, string.Empty);
+
+        public static string NormalizeLineEndings(this string value)
+        {
+            value = value.Replace("\r\n", "\n");
+            value = value.Replace('\r', '\n');
```
Comment (stephentoub) [inline src/System.Text.Json/tests/JsonTestHelper.cs]: Why is this second line needed?

File: src/System.Text.Json/tests/JsonElementWriteTests.cs [hunk 2]
```diff
@@ -1206,13 +1206,13 @@ private static void WriteSimpleValue(bool indented, string jsonIn, string jsonOu
...
             Assert.Equal(
-                expectedValue,
+                expectedValue.NormalizeLineEndings(),
                 Encoding.UTF8.GetString(
                     buffer.WrittenSpan
 #if netfx
                         .ToArray()
 #endif
-                    ));
+                    ).NormalizeLineEndings());
```
Comment (stephentoub) [inline src/System.Text.Json/tests/JsonElementWriteTests.cs]: Couldn't this end up hiding bugs where, for example, the writer ends up using "\n" on Windows?

File: src/System.Text.Json/tests/JsonTestHelper.cs (hunk 1 above)
Comment (krwq) [reply src/System.Text.Json/tests/JsonTestHelper.cs]: In case you checked out on OS which uses CR as line ending

Comment (stephentoub) [reply src/System.Text.Json/tests/JsonTestHelper.cs]: What OSes that developers actively use to run corefx tests use "\r" as a line ending?

File: src/System.Text.Json/tests/JsonElementWriteTests.cs (hunk 2 above)
Comment (krwq) [reply src/System.Text.Json/tests/JsonElementWriteTests.cs]: I'll create a second copy of AssertContent which normalizes and verify all callsites - this way we will avoid any potential issues now and in the future (name should be clearer what it does)

File: src/System.Text.Json/tests/JsonTestHelper.cs (hunk 1 above)
Comment (krwq) [reply src/System.Text.Json/tests/JsonTestHelper.cs]: actually reading now, I'm not sure if there is such OS nowadays

File: src/System.Text.Json/tests/JsonElementWriteTests.cs (hunk 2 above)
Comment (krwq) [reply src/System.Text.Json/tests/JsonElementWriteTests.cs]: ended up doing it directly on each test as it was more clearer what's going on

File: src/System.Text.Json/tests/JsonTestHelper.cs (hunk 1 above)
Comment (krwq) [reply src/System.Text.Json/tests/JsonTestHelper.cs]: seems some of the tests already do normalization slightly differently so I have generalized the logic

**Comment by krwq** [comment]
I have force-pushed to test with latest (also code is almost completely different so no benefit of having iteration)

File: src/System.Text.Json/tests/JsonElementWriteTests.cs [hunk 3]
```diff
@@ -207,19 +201,19 @@ public static void WriteSimpleArray(bool indented)
...
   0,
   1
-]",
+]".NormalizeLineEndings(),
                 "[2,4,6,0,1]");
         }
 
         [Theory]
-        [InlineData(false)]
+        //[InlineData(false)]
```
Comment (krwq) [inline src/System.Text.Json/tests/JsonElementWriteTests.cs]: oops

**Comment by ahsonkhan** [comment]
Is this change really worth it to add complexity to the test code? I am not convinced you if this change provides much value. Are there other projects that have this type of check?

cc @steveharter, @bartonjs - what do you think. Have you ever had to deal with this issue, locally?

**Comment by krwq** [comment]
optionally we can do something similar to this:
https://github.com/dotnet/corefx/blob/master/.gitattributes#L67

**Comment by bartonjs** [comment]
Yep, it's also a problem because `@""` strings embed the host OS newline.  So JsonElementWriteTests has some stuff like https://github.com/dotnet/corefx/blob/8628de7f486c825b6055ac85feb2c3f4f3981c75/src/System.Text.Json/tests/JsonElementWriteTests.cs#L12-L22

It's a rational change to make (forcing line-endings in gitattributes is not)

**Comment by ericstj** [comment]
Moved this PR to future since issue it is fixing is moved to future.

**Comment by eiriktsarpalis** [comment]
@krwq what's the status of this PR?

**Comment by krwq** [comment]
I've talked with @ahsonkhan offline and he decided he wants to take this so will need to rebase and fix merge conflicts here

File: src/System.Text.Json/tests/JsonElementWriteTests.cs [hunk 4]
```diff
@@ -1192,13 +1158,6 @@ private static void WriteSimpleValue(bool indented, string jsonIn, string jsonOu
                 writer.WriteEndObject();
                 writer.Flush();
 
-                if (indented && s_replaceNewlines)
-                {
-                    AssertContents(
```
Comment (ahsonkhan) [inline src/System.Text.Json/tests/JsonElementWriteTests.cs]: Are we losing any coverage when removing these?

Comment (krwq) [reply src/System.Text.Json/tests/JsonElementWriteTests.cs]: no, this is already checked - in some cases we have previously double checked the same assert

**Pull request merged**